### PR TITLE
refactor(csharp): move, tiny additions and cleanings

### PR DESCRIPTION
**Pull request by morganleroi**
## 🧭 What and Why

Don't be afraid by the files changed counter. ⚠️ 

No behaviour changes, only refactorings. ⚠️ 

🎟 JIRA Ticket: [DI-1811](https://algolia.atlassian.net/browse/DI-1811)

### Changes included:
- Move Echo Requester into Test assembly
- Rework Logs level (and add some).
- Clean using
- Adapt Readme for V7
- Reset Alpha Version
- Refactor ClientUtil and QueryHelper.

[DI-1811]: https://algolia.atlassian.net/browse/DI-1811?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs [hunk 1]
```diff
@@ -5,7 +5,6 @@
 using Algolia.Search.Serializer;
 using Algolia.Search.Transport;
 using Algolia.Search.Utils;
-using Microsoft.Extensions.Logging;
```
Comment (millotp) [inline clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs:8]: Is there a linter to remove unused imports ?

**Review by millotp — COMMENTED** [review]
Nice cleanup ! Can you merge the changelog of you reset the version pls ?

File: clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs (hunk 1 above)
Comment (morganleroi) [reply clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs:8]: Dotnet format do not remove unused using unfortunately ... I have to use other linter and I'm not a fan of the formatting. So far, I'll this as is

Comment (shortcuts) [reply clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs:8]: Had the same case for python and I'm just running the one that removes the unused stuff before the formatter

idk if applicable here but worth checking

Comment (morganleroi) [reply clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs:8]: If I chain `csharpier` then `dotnet format` the result is not so good. `using` are removed but the formatting is completely influenced by `csharpier` rules (that cannot be override).

Comment (morganleroi) [reply clients/algoliasearch-client-csharp/algoliasearch/Clients/AlgoliaConfig.cs:8]: I think I'll let the formatting like that.

**Comment by morganleroi** [comment]
@millotp  I removed the version bump + changelog. I'll do it in the release PR.

**Pull request merged**